tests/e2e/util.go

```diff
@@ -16,3 +31,207 @@ func ReadModuleTemplate(filepath string) (*v1beta2.ModuleTemplate, error) {
 	err = yaml.Unmarshal(moduleFile, &moduleTemplate)
 	return moduleTemplate, err
 }
+
+func ExecuteKymaDeployCommand() error {
+	deployCmd := exec.Command("kyma", "alpha", "deploy")
+	deployOut, err := deployCmd.CombinedOutput()
+	if err != nil {
+		return errKymaDeployCommandFailed
+	}
+
+	if !strings.Contains(string(deployOut), "Kyma CR deployed and Ready") {
+		return errKymaDeployCommandFailed
+	}
+
+	return nil
+}
+
+func DeploymentIsReady(ctx context.Context,
+	k8sClient client.Client,
+	deploymentName string,
+	namespace string) bool {
+	var deployment appsv1.Deployment
+	err := k8sClient.Get(ctx, client.ObjectKey{
+		Namespace: namespace,
+		Name:      deploymentName,
+	}, &deployment)
+
+	GinkgoWriter.Println("Available replicas:", deployment.Status.AvailableReplicas)
+	if err != nil || deployment.Status.AvailableReplicas == 0 {
+		return false
+	}
+
+	return true
+}
+
+func KymaCRIsInReadyState(ctx context.Context,
+	k8sClient client.Client,
+	kymaName string,
+	namespace string) bool {
+	var kyma v1beta2.Kyma
+	err := k8sClient.Get(ctx, client.ObjectKey{
+		Namespace: namespace,
+		Name:      kymaName,
+	}, &kyma)
+
+	if err != nil || kyma.Status.State != v1beta2.StateReady {
+		return false
+	}
+
+	return true
+}
+
+func ApplyModuleTemplate(
+	moduleTemplatePath string) error {
+	cmd := exec.Command("kubectl", "apply", "-f", moduleTemplatePath)
+
+	_, err := cmd.CombinedOutput()
+	if err != nil {
+		return errModuleTemplateNotApplied
+	}
+
+	return nil
+}
+
+func enableKymaModuleWithExpectedExitCode(moduleName string, expectedExitCode int) error {
+	cmd := exec.Command("kyma", "alpha", "enable", "module", moduleName, "-w")
+	err := cmd.Run()
+	var exitCode int
+	if exitErr, ok := err.(*exec.ExitError); ok {
+		exitCode = exitErr.ExitCode()
+	} else {
+		exitCode = cmd.ProcessState.ExitCode()
+	}
+
+	GinkgoWriter.Println("Exit code", exitCode)
+	if exitCode != expectedExitCode {
+		return errModuleEnablingFailed
+	}
+	return nil
+}
+
+func EnableKymaModuleWithReadyState(moduleName string) error {
+	expectedExitCode := 0
```

Comment:
I see no benefit of defining a local variable in here.

I suggest either this:
```go
return enableKymaModuleWithExpectedExitCode(moduleName, 0)
```

or this:
```go
const {
    exitCodeNoError = 0
    exitCodeError   = 2
}

...
func EnableKymaModuleWithReadyState(moduleName string) error {
    return enableKymaModuleWithExpectedExitCode(moduleName, exitCodeNoError)
}

func EnableKymaModuleWithWarningState(moduleName string) error {
    return enableKymaModuleWithExpectedExitCode(moduleName, exitCodeError)
}
```